### PR TITLE
Implanted foreign bodyparts will resist being removed on species change

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

Title.
Also, to make bodypart code slightly nicer, I retooled some variables to be part of a new bitfield called bodypart_flags.

## Why It's Good For The Game

We've been trying to move away from the species datum for limb stuff precisely because of funny shenanigans like this, no?

## Changelog

:cl:
refactor: Implanted foreign limbs will no longer be wiped by species change.
/:cl:

### CODE REVIEW

**Comment by ChungusGamer666** [comment]
not sure if this counts as a feature or refactor tbh

File: code/modules/surgery/prosthetic_replacement.dm [hunk 1]
```diff
@@ -120,12 +122,13 @@
 			span_notice("[user] finishes the attachment procedure!"),
 		)
 		display_pain(target, "You feel a strange sensation from your new [parse_zone(target_zone)].", TRUE)
-		qdel(tool)
```
Comment (san7890) [inline code/modules/surgery/prosthetic_replacement.dm:123]: lol wtf

Comment (san7890) [reply code/modules/surgery/prosthetic_replacement.dm:123]: wacky ass shit

**Review by Time-Green — APPROVED** [review]

**Comment by ChungusGamer666** [comment]
i got an email about this being merged!!!

**Pull request merged**